invoke/runners.py

```diff
@@ -659,6 +659,8 @@ def handle_stdin(self, input_, output, echo):
                 elif data is not None:
                     # When reading from file-like objects that aren't "real"
                     # terminal streams, an empty byte signals EOF.
+                    if not self.using_pty:
+                        self._close_proc_stdin()
```

Comment:
can check for the existence of attribute on self to allow subclasses to defer the fix